src/OpenSage.Game/Logic/Object/Locomotor.cs

```diff
@@ -124,14 +155,33 @@ public void MoveTowardsPosition(in TimeInterval gameTime, in Vector3 targetPoint
 
             // The distance we're moving
             var direction = Vector2.Normalize(delta.Vector2XY());
-            trans += new Vector3(direction * distance, 0.0f);
+            var moveDirection = direction;
+
+            // Calculate rotation
+            var currentYaw = -transform.EulerAngles.Z;
+
+            var targetYaw = MathUtility.GetYawFromDirection(direction);
+            var angleDelta = MathUtility.CalculateAngleDelta(targetYaw, currentYaw);
+
+            var d = MathUtility.ToRadians(GetTurnRate()) * deltaTime;
+            var newDelta = -MathF.Sign(angleDelta) * MathF.Min(MathF.Abs(angleDelta), MathF.Abs(d));
+            var yaw = currentYaw + newDelta;
+            _gameObject.Yaw = Math.Clamp(newDelta / deltaTime, MathUtility.ToRadians(-GetFrontWheelTurnAngle()), MathUtility.ToRadians(GetFrontWheelTurnAngle()));
 
             switch (_locomotorTemplate.Appearance)
             {
                 case LocomotorAppearance.Thrust:
                     trans.Z += (distance / distanceRemaining) * (targetPoint.Z - trans.Z);
                     break;
+                case LocomotorAppearance.Treads:
+                    if (MathF.Abs(targetYaw - yaw) > MathUtility.ToRadians(2.0f)) //first fully rotate towards target point
```

Comment:
Vehicles with treads can't move forward while turning

Reply:
I attempted to implement this in January, and got nowhere. Thanks for making progress on it 😄